spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/actions/BaseRewriteDataFilesSparkAction.java

```diff
@@ -327,6 +327,7 @@ private Result doExecuteWithPartialProgress(RewriteExecutionContext ctx, Stream<
       LOG.error("{} is true but no rewrite commits succeeded. Check the logs to determine why the individual " +
           "commits failed. If this is persistent it may help to increase {} which will break the rewrite operation " +
           "into smaller commits.", PARTIAL_PROGRESS_ENABLED, PARTIAL_PROGRESS_MAX_COMMITS);
+      throw new CommitFailedException("No rewrite commits succeeded after max commits");
```

Comment:
The issue here is that it may not have been a commit failed exception, it could have been a data file permission issue, out of disk space issue, or any other host of problems. So we don't actually know if there was a commit failure, we just know that none of the commits succeeded. I think if you wanted we could still throw an error here but I don't believe it should be a commit failed exception.

Reply:
Hey, @RussellSpitzer thanks for your review! I got your point, I think we can throw a new NoCommitSucceedException which extends RuntimeException here. What do you think of this?